tt_eager/tt_dnn/op_library/composite/composite_ops.cpp

```diff
@@ -831,19 +832,38 @@ Tensor _addcdiv(
     Tensor result = add(input_a, t_factor, std::nullopt, output_mem_config);
     Tensor t_inf = full_like(input_a, std::numeric_limits<float>::infinity(), output_mem_config);
     Tensor t_nan = full_like(input_a, std::nanf(""), output_mem_config);
-    return where(eqz(input_c, output_mem_config),
+
+    if(output_tensor.has_value())
+    {
+        output_tensor = where(eqz(input_c, output_mem_config),
```

Comment:
In this case, isn't the storage member variable of the existing output_tensor deallocated?

Reply:
I don't think so, if the output tensor is passed by the user, the result should be stored in output tensor else we may create a new tensor.

Reply:
It’s not stored but replaced. If it’s working correctly, t2 and output_tensor should be the same.

```
> /home/ubuntu/tt-metal/tests/tt_eager/python_api_testing/sweep_tests/tt_lib_ops.py(589)eltwise_addcdiv_with_output()
-> return tt2torch_tensor(t3)
(Pdb) l
584  	    t2 = setup_tt_tensor(z, device, layout[2], input_mem_config[2], dtype[2])
585  	    output_tensor = setup_tt_tensor(z, device, layout[2], input_mem_config[2], dtype[2])
586  	    t3 = ttl.tensor.addcdiv(t0, t1, t2, scalar, output_mem_config, output_tensor)
587  	    breakpoint()
588
589  ->	    return tt2torch_tensor(t3)
590
591
592  	@setup_host_and_device
593  	def unary_div_bw(
594  	    x,  # grad_tensor
(Pdb) output_tensor
ttnn.Tensor([[[[-41.00000, -88.50000,  ..., -42.50000, -28.62500],
               [26.00000, -14.81250,  ..., -53.50000, 61.75000],
               ...,
               [-44.50000, 39.00000,  ..., -23.00000, 32.00000],
               [76.50000, 69.00000,  ..., -7.40625, 70.50000]]]], shape=Shape([1, 1, 32, 32]), dtype=DataType::BFLOAT16, layout=Layout::TILE)
(Pdb) t3
ttnn.Tensor([[[[-28.50000, -39.25000,  ..., -28.00000, 28.25000],
               [71.00000, 56.25000,  ...,  3.60938, 11.68750],
               ...,
               [17.62500, -22.00000,  ..., 10.25000, -10.31250],
               [-2.95312, -17.62500,  ..., -13.50000, 13.81250]]]], shape=Shape([1, 1, 32, 32]), dtype=DataType::BFLOAT16, layout=Layout::TILE)
```

Reply:
At the very least, the last op in the composite op should take the output tensor parameter and store the result in it. CC: @eyonland